### DISCUSSION TITLE
Is there a extension/plugin/hook for wikipedia-like tables?

### DISCUSSION THREAD

**Discussion by Andre601** [mkdocs/mkdocs#3459] [Q&A] [answered]
I'm sure you know these tables Wikipedia has on their different pages, that shows some nice info...

I would like to do something similar within MkDocs. Tho, the only way I know to do it is using HTML itself, which is rather ugly to look at.

So I was wondering if there is a markdown extension, plugin or python hook that would implement a way for an easy to look at table in markdown that renders as a HTML table in the end...

The only plugin I know of is neoteroi's spantable (As part of the mkdocs-plugins plugin) but I don't like using it because A) it has this extra `::spantable::` thing that I find weird to have, B) I have to add extra CSS which I feel like is weird because the theme I use already does this (Probs because the table isn't wrapped into the usual divs with classes it normally has?) and C) the entire plugin seems unmaintained, so I would preferably use something that is updated.

Is there anything?

FYI, I talk about having an easy syntax to add something along the lines of this to a page (Used the wikipedia page about mkdocs as example):
![grafik](https://github.com/mkdocs/mkdocs/assets/11576465/2f2ca77f-cc8c-44ca-af4a-905cd399cd97)

**Comment by pawamoy** [comment, marked as answer]
With [Material for MkDocs inline admonitions](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#inline-blocks) and a bit of re-styling:

```md
<style>
    .admonition.wiki thead {
        display: none;
    }
    .admonition.wiki table {
        border: 0 !important;
    }
    .admonition.wiki tr, .admonition.wiki td {
        border: 0 !important;
        border-top: 0 !important;
        padding: 2px !important;
    }
</style>
```
```md
!!! wiki inline end ""

    ![mkdocs logo](https://requisitos-de-software.github.io/2023.1-Letterboxd/assets/logo-mkdocs.png)

    Info            | Value
    --------------- | -----
    Developer(s)    | Tom Christie, ...
    Initial release | January 24, 2014
    Stable release  | 1.5.3 ...
    Repository      | https://github.com/mkdocs/mkdocs
    ...             | ...
```

![wiki](https://github.com/mkdocs/mkdocs/assets/3999221/d94a4113-83fc-4b00-bedf-ca4748d30b09)

Probably possible with other themes and other admonition extensions too (markdown-callouts, pymdownx.blocks.admonition), you just need to float the admonition on the right.

**Reply by pawamoy** [reply]
Yep:

```md
<style>
    .admonition.wiki {
        float: right;
        max-width: 300px;
        position: relative;
        top: -20px;
        margin-left: 20px;
    }
    .admonition.wiki thead {
        display: none;
    }
    .admonition.wiki table {
        border: 0 !important;
    }
    .admonition.wiki tr, .admonition.wiki td {
        border: 0 !important;
        border-top: 0 !important;
        padding: 2px !important;
    }
</style>
```

```md
> WIKI: **MkDocs**
>
> ![mkdocs logo](https://requisitos-de-software.github.io/2023.1-Letterboxd/assets/logo-mkdocs.png)
>
> Info            | Value
> --------------- | -----
> Developer(s)    | Tom Christie, ...
> Initial release | January 24, 2014
> Stable release  | 1.5.3 ...
> Repository      | https://github.com/mkdocs/mkdocs
> ...             | ...
```

**Reply by oprypin** [reply]
Amazing answer!

**Reply by Andre601** [reply]
Yeah. Tho, need to look how to implement this in pymdownx's admonition extension.

**Reply by oprypin** [reply]
(wow [that extension](https://facelessuser.github.io/pymdown-extensions/extensions/blocks/plugins/admonition/) is even uglier than I remember)

@Andre601 You could just use HTML without any admonition extension like in my answer but without my CSS.

```markdown
<div class="admonition wiki inline end" markdown="1">

![mkdocs logo](https://upload.wikimedia.org/wikipedia/commons/d/dd/MkDocs_Logo.png)

Info            | Value
--------------- | -----
...             | ...

</div>
```

**Reply by facelessuser** [reply]
> (wow [that extension](https://facelessuser.github.io/pymdown-extensions/extensions/blocks/plugins/admonition/) is even uglier than I remember)

Uh...thanks?

**Reply by pawamoy** [reply]
I suppose @oprypin does not like the syntax :smiling_face_with_tear:

**Reply by Andre601** [reply]
Let's not get too off-topic here... (Tho, I personally would've loved the `:::` syntax rather than `///` but that's just how it is)

**Reply by Andre601** [reply]
It *kinda* works btw...

One issue I currently encounter is, that the generated admonition display the `p` title object, no matter what I try.

The main issue here is, that the pymdownx version has a different way of defining a title, which doesn't use `""`, therefore is there no clear way to define an empty title. I doubt it even has the feature of `title empty -> Don't include title` (yet).

So I need to find a workaround... Maybe facelessuser has any input on this?

Another issue I encounter is, that the headers of the table are still readable... How are they hidden in your example? I've used the same CSS and they show for me.

To give a comparison, left is the blocks admonition and right is the original ones...
![grafik](https://github.com/mkdocs/mkdocs/assets/11576465/77b2afeb-6e74-4193-b2d9-19d167b5cc36)

**Reply by pawamoy** [reply]
I didn't try with the blocks admonition extension. I would suggest to check the generated HTML and adapt the CSS :shrug:

Also, even if you can't pass an empty title, you can hide it with CSS too. That's what I did for the table headers, since we can't declare a table without headers.

**Reply by Andre601** [reply]
Yeah, that works

**Reply by facelessuser** [reply]
> The main issue here is, that the pymdownx version has a different way of defining a title, which doesn't use "", therefore is there no clear way to define an empty title. I doubt it even has the feature of title empty -> Don't include title (yet).

```
/// admonition |
content
///
```

> I suppose @oprypin does not like the syntax 🥲

Yeah, it's impossible to please everyone, but there are nicer ways to say things 😬 .

**Reply by Andre601** [reply]
> ```
> /// admonition |
> content
> ///
> ```

Doesn't work, as I mentioned. It turns it into an admonition with the type as the title (As if you didn't define a title at all).

**Reply by pawamoy** [reply]
Same with markdown-callouts, empty title means the type becomes the title.

**Reply by facelessuser** [reply]
@Andre601 it works only for `admonition`, not for `note`, etc. With that said, this is something that we can easily fix.

**Reply by facelessuser** [reply]
PR with fix https://github.com/facelessuser/pymdown-extensions/pull/2235

**Reply by facelessuser** [reply]
The latest release now supports empty titles via `/// note |`, etc.

**Comment by oprypin** [comment]
It is quite a common idea to reuse some existing HTML elements that have existing Markdown syntax for them, and then add a ton of CSS for them. For this example it might be something like a table or a definition list.

---

A slight variation based on @pawamoy's answer, to not rely on mkdocs-material and admonitions that need to be indented.

```yaml
markdown_extensions:
  - tables
  - md_in_html

extra_css:
  - a_file_that_contains_the_following.css
```
```css
.infobox {
    /* Stolen from Wikipedia's CSS! */
    border: 1px solid #a2a9b1;
    background-color: #f8f9fa;
    margin: 0.5em 0 0.5em 1em;
    padding: 0.2em;
    float: right;
    clear: right;
    font-size: 88%;
    line-height: 1.5em;
    width: 22em;
}
.infobox p:first-child {
    text-align: center;
}
.infobox thead {
    display: none;
}
.infobox tr, .infobox td {
    border: 0;
    padding: 2px 5px;
}
.infobox td:first-child {
    font-weight: bold;
}
```



Page body:
```markdown
<div class="infobox" markdown>

![mkdocs logo](https://upload.wikimedia.org/wikipedia/commons/d/dd/MkDocs_Logo.png)

Info            | Value
--------------- | -----
Developer(s)    | Tom Christie, ...
Initial release | January 24, 2014
Stable release  | 1.5.3 ...
Repository      | https://github.com/mkdocs/mkdocs
...             | ...

</div>
```


Result, in `mkdocs` theme:

![image](https://github.com/mkdocs/mkdocs/assets/371383/f145a35e-1713-409c-8e4c-0329cfce2988)

---

There's also another example of something like this in action
https://squidfunk.github.io/mkdocs-material/reference/grids/#using-card-grids
It **very heavily** reuses existing syntax to display a new kind of component

**Reply by Andre601** [reply]
I actually do use mkdocs-material, so it's not an issue for me. The biggest problem is, that I'm using the adminition blocks from Pymdownx's blocks extension. And I'm not sure how I would apply the `inline end` to it.

**Reply by Andre601** [reply]
Also, I'm not a fan of using HTML within markdown here... Even if it is just a div.

**Comment by redtide** [comment]
As a side note, only for the MKDocs theme using Bootstrap: what about a "wikipage.j2" page template using a Bootstrap' aside card?
So fill the card/table information from the page' frontmatter/metadata instead write manually a markdown table in its content each time, and markdown only for the content on the left.
